rclcpp/include/rclcpp/node_options.hpp

```diff
@@ -358,6 +372,8 @@ class NodeOptions
     rclcpp::QoSInitialization::from_rmw(rmw_qos_profile_parameter_events)
   );
 
+  rclcpp::QoS rosout_qos_ = rclcpp::QoS(rcl_qos_profile_rosout_default);
```

Comment:
```suggestion
  rclcpp::QoS rosout_qos_ = rclcpp::QoS(QoSInitialization::from_rmw(rcl_qos_profile_rosout_default), rcl_qos_profile_rosout_default);
```

Reply:
Ok

Reply:
> This was intentionally not added before.

So what is the disadvantages you want to avoid ?

Reply:
I think that the rationale was to force users to pass the history depth.
@wjwwood may recall better.

Reply:
Yeah, you should not be able to create a `QoS` without arguments or with just an rmw struct.

Reply:
I would recommend that you do not change the implementation of `rclcpp::QoS` nor inherit from `rclcpp::QoSInitialization`, but rather inherit from `rclcpp::QoS` as is the case with the profiles like `rclcpp::SensorDataQoS`.